src/ota/ota_handler.rs

```diff
@@ -233,21 +233,53 @@ impl<'a> OTAHandler<'a> {
         if self.state_repository.exists() {
             info!("Found pending update");
             let state = self.state_repository.read()?;
+            let result_response = match self.do_pending_ota(&state).await {
+                Ok(()) => {
+                    info!("OTA successful");
+                    self.send_ota_response(sdk, &state.uuid, OTAStatus::Done)
+                        .await
+                }
+                Err(error) => {
+                    warn!("OTA failed, error -> {:?}", error);
+                    self.send_ota_response(sdk, &state.uuid, OTAStatus::Error(OTAError::Failed))
+                        .await
+                }
+            };
+
+            if result_response.is_err() {
+                warn!(
```

Comment:
Could this warn be moved in the `Err` branch of the match above?

Reply:
no, because is the result of `self.send_ota_response` returned by the match block.